hack/make-rules/test-cmd-util.sh

```diff
@@ -76,6 +76,43 @@ daemonsets="daemonsets"
 controllerrevisions="controllerrevisions"
 
 
+# include shell2junit library
+sh2ju="${KUBE_ROOT}/third_party/forked/shell2junit/sh2ju.sh"
+if [[ -f "${sh2ju}" ]]; then
+  source "${sh2ju}"
+else
+  echo "fetching sh2ju.sh from GitHub"
+  source <(curl -fsS --retry 3 'https://raw.githubusercontent.com/kubernetes/kubernetes/master/third_party/forked/shell2junit/sh2ju.sh')
+fi
+
+# record_command runs the command and records its output/error messages in junit format
+# it expects the first to be the name of the command
+# Example:
+# record_command run_kubectl_tests
+#
+# WARNING: Variable changes in the command will NOT be effective after record_command returns.
+#          This is because the command runs in subshell.
+function record_command() {
+    set +o nounset
+    set +o errexit
```

Comment:
I'm trying to do something similar to https://github.com/kubernetes/test-infra/blob/b72c1687c3b71a44bf9953447670fb566ed53f5e/jenkins/e2e-image/e2e-runner.sh#L58-L82.
I will check if there is a fix in [upstream](https://github.com/manolo/shell2junit). If there is, we can bump it?